src/dbnode/persist/fs/commitlog/files.go

```diff
@@ -27,20 +27,54 @@ import (
 	"sort"
 	"time"
 
+	"github.com/m3db/m3/src/dbnode/persist"
 	"github.com/m3db/m3/src/dbnode/persist/fs"
 	"github.com/m3db/m3/src/dbnode/persist/fs/msgpack"
 )
 
-// File represents a commit log file and its associated metadata.
-type File struct {
-	FilePath string
-	Start    time.Time
-	Duration time.Duration
-	Index    int64
+var (
+	timeNone = time.Unix(0, 0)
+)
+
+// NextFile returns the next commitlog file.
+func NextFile(opts Options) (string, int, error) {
+	files, _, err := Files(opts)
+	if err != nil {
+		return "", -1, err
+	}
+
+	newIndex := 0
+	for _, f := range files {
+		if int(f.Index) >= newIndex {
+			newIndex = int(f.Index + 1)
+		}
+	}
+
+	for ; ; newIndex++ {
+		var (
+			prefix = opts.FilesystemOptions().FilePathPrefix()
+			// We pass timeNone for the commitlog file block start because that field
+			// is no longer required and can just be zero.
+			// TODO(rartoul): It should actually be completely backwards compatible to
+			// change the commitlog filename structure (because we don't rely on the name
+			// for any information, we just list all the files in a directory and then
+			// read their encoded heads to obtain information about them), so in the future
+			// we can just get rid of this.
```

Comment:
Can you add a `TODO(v1): remove this when ok to be backwards incompatible`

Reply:
I think this is just a regular TODO because it is backwards compatible (existing code doesn't rely on filename). I just didn't want to make the change in this P.R

Reply:
Sure thing.